src/lib.rs

```diff
@@ -97,6 +103,13 @@ pub fn init() -> Result<(), ()> {
     }
 }
 
+#[ctor]
+fn init_real() {
+    unsafe {
+        ffi::sodium_init();
```

Comment:
@mmastrac can we fail here somehow?

Reply:
As this is a pure FFI call, it's a safe way to use `#[ctor]`. If your users downstream try to use some of the bindings in their own `#[ctor]` code they will obviously need to be cautious as there's no guarantee to the initialization order (more of a doc issue/caveat).